libsolidity/formal/ModelCheckerSettings.h

```diff
@@ -173,7 +173,10 @@ struct ModelCheckerSettings
 	bool showUnsupported = false;
 	smtutil::SMTSolverChoice solvers = smtutil::SMTSolverChoice::Z3();
 	ModelCheckerTargets targets = ModelCheckerTargets::Default();
-	std::optional<unsigned> timeout;
+	// Default timeout per query in milliseconds.
+	// We keep it as optional because tests may overwrite this to use
+	// resource limit for reproducibility.
+	std::optional<unsigned> timeout = 1000;
```

Comment:
I've read the above comment, but I still don't understand why this is optional. What does `tests may override this to use resource limit` mean? Will the tests be overriding the `std::optional` somehow, or will they just change the actual value of the timeout with some number?